src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -133,18 +133,14 @@ public static Tensor<T> ConcatenateOnDimension<T>(int dimension, params scoped R
                 ThrowHelper.ThrowArgument_ConcatenateTooFewTensors();
 
             if (dimension < -1 || dimension > tensors[0].Rank)
-                ThrowHelper.ThrowArgument_InvalidAxis();
+                ThrowHelper.ThrowArgument_InvalidDimension();
 
-            // Calculate total space needed.
-            nint totalLength = 0;
-            for (int i = 0; i < tensors.Length; i++)
-                totalLength += tensors[i].FlattenedLength;
+            Tensor<T> tensor;
 
-            nint sumOfAxis = 0;
             // If axis != -1, make sure all dimensions except the one to concatenate on match.
             if (dimension != -1)
             {
-                sumOfAxis = tensors[0].Lengths[dimension];
+                nint sumOfAxis = tensors[0].Lengths[dimension];
```

Comment:
Does this need to check for overflow?

Reply:
this is the sum of the axis that we are concatenating. do you think that will ever go over `nint`?

Reply:
Realistically it *shouldn't*, but a user definitely could make it.

For example, just run on 32-bit with a 128MB tensor and then try to stack it with itself 32 times.